### ISSUE TITLE
Compatibility with `docutils>=0.19`

### ISSUE THREAD

**Issue by amotl** [crate/sphinx_csv_filter#41]
Hi. With 7faeb4f, we limited runtime dependencies to `docutils<0.19`. When upgrading, the tests croak like this.

```python
______ ERROR collecting tests/test_csv.py ______
tests/test_csv.py:49: in <module>
    directive = CSVFilterDirective(
.venv/lib/python3.11/site-packages/docutils/parsers/rst/__init__.py:331: in __init__
    self.reporter = state_machine.reporter
E   AttributeError: 'NoneType' object has no attribute 'reporter'
```

**Comment by Irockasingranite** [comment]
Hi! I've been trying to package `sphinx_csv_filter` for use in a larger environment, and the `docutils` dependency limit has caused minor pain.

I took a look at the failing tests, and they seem to crash only because of an issue with the constructor arguments: The `Directive` constructor was changed to actually access one of its arguments (`state_machine.reporter`), but the tests pass all `None`s to it.

If I change those dummy arguments to something like
```python
class Dummy:
    def __init__():
        self.reporter = None

directive = CSVFilterDirective(
    None, None, None, None, None, None, None, None, Dummy())
```
all the tests pass again.

If the failing tests are the only reason for the dependency bound, I suspect it's actually not necessary.
